samples/test/cache_v2.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2021 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Random Two step v2-compatible pipeline."""
```

Comment:
is this sample any different from two_step sample?

I think it's easier to maintain if we add extra test cases for two_step sample that verifies caching, compared to adding a new not-so-different sample.
(you can also maintain a separate two_step_cache_test.py file for your own test cases that import the two_step pipeline if you want a separate file)

Reply:
Yes. It is different because it needs random inputs.

Reply:
I reused the train_op which is the same as two_step

Reply:
done